batcher/batcher_test.go

```diff
@@ -0,0 +1,105 @@
+package batcher
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdb/influxdb"
+)
+
+// TestBatch_Size ensures that a batcher generates a batch when the size threshold is reached.
+func TestBatch_Size(t *testing.T) {
+	batchSize := 5
+	batcher := NewBatcher(batchSize, time.Hour)
+	if batcher == nil {
+		t.Fatal("failed to create batcher for size test")
+	}
+
+	in := make(chan influxdb.Point)
+	out := make(chan []influxdb.Point)
+
+	go batcher.Start(in, out)
+
+	var p influxdb.Point
+	for i := 0; i < batchSize; i++ {
```

Comment:
OK, yeah, I thought that was what you were getting at.

I think it's OK -- I did this on purpose. The point of this test is to ensure that if events go into the batcher, the events come out as expected. Obviously in real client code it would use goroutines, and reads of the `out` channel would block if nothing is expected, but I don't want to test that.